llvm/lib/Transforms/Scalar/GVNSink.cpp

```diff
@@ -535,14 +535,10 @@ class ValueTable {
     uint32_t e = ExpressionNumbering[exp];
     if (!e) {
       hash_code H = exp->getHashValue([=](Value *V) { return lookupOrAdd(V); });
-      auto I = HashNumbering.find(H);
-      if (I != HashNumbering.end()) {
-        e = I->second;
-      } else {
-        e = nextValueNumber++;
-        HashNumbering[H] = e;
-        ExpressionNumbering[exp] = e;
-      }
+      auto [I, Inserted] = HashNumbering.try_emplace(H, nextValueNumber + 1);
```

Comment:
Shouldn't this be?
```suggestion
      auto [I, Inserted] = HashNumbering.try_emplace(H, nextValueNumber);
```
The old code is using the result of post-inc, which is the pre-increment value.

Reply:
Oops.  Thanks for catching this!  Fixed in the latest iteration.